testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -187,6 +187,162 @@ public void testReadSparkCreatedTable(String sparkTableFormat, String expectedTr
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS}, dataProvider = "testReadSparkCreatedTableDataProvider")
+    public void testReadTrinoCreatedTable(String sparkTableFormat, String expectedTrinoTableFormat)
+    {
+        String sparkTableName = "trino_created_table_" + sparkTableFormat.replaceAll("[^a-zA-Z]", "").toLowerCase(ENGLISH) + "_" + randomTableSuffix();
+        String trinoTableName = format("%s.default.%s", TRINO_CATALOG, sparkTableName);
+
+        onTrino().executeQuery("SET SESSION hive.timestamp_precision = 'NANOSECONDS'");
+        onTrino().executeQuery(format(
+                "CREATE TABLE %s ( " +
+                        "   a_boolean boolean, " +
+                        "   a_tinyint tinyint, " +
+                        "   a_smallint smallint, " +
+                        "   an_integer integer, " +
+                        "   a_bigint bigint, " +
+                        "   a_real real, " +
+                        "   a_double double, " +
+                        "   a_short_decimal decimal(11, 4), " +
+                        "   a_long_decimal decimal(26, 7), " +
+                        "   a_string varchar, " +
+                        // TODO binary
+                        "   a_date date, " +
+                        "   a_timestamp_seconds timestamp(9), " +
+                        "   a_timestamp_millis timestamp(9), " +
+                        "   a_timestamp_micros timestamp(9), " +
+                        "   a_timestamp_nanos timestamp(9), " +
+                        // TODO interval
+                        // TODO array
+                        // TODO struct
+                        // TODO map
+                        "   a_dummy varchar " +
+                        ") " +
+                        "WITH ( " +
+                        "   format = '%s' " +
+                        ")",
+                trinoTableName,
+                expectedTrinoTableFormat));
+
+        // nulls
+        onTrino().executeQuery("INSERT INTO " + trinoTableName + " VALUES (" + join(",", nCopies(16, "NULL")) + ")");
```

Comment:
This test should be run with the default Parquet writer and the experimental too.